test/e2e/apps/daemon_restart.go

```diff
@@ -52,27 +52,30 @@ const (
 	restartTimeout      = 10 * time.Minute
 	numPods             = 10
 	sshPort             = 22
-	ADD                 = "ADD"
-	DEL                 = "DEL"
-	UPDATE              = "UPDATE"
+	// ADD represents for ADD event
+	ADD = "ADD"
+	// DEL represents for DEL event
+	DEL = "DEL"
+	// UPDATE represents for UPDATE event
+	UPDATE = "UPDATE"
 )
 
-// restartDaemonConfig is a config to restart a running daemon on a node, and wait till
+// RestartDaemonConfig is a config to restart a running daemon on a node, and wait till
 // it comes back up. It uses ssh to send a SIGTERM to the daemon.
-type restartDaemonConfig struct {
+type RestartDaemonConfig struct {
```

Comment:
private struct with private methods should be fine.
what was the linter error here?

Reply:
Thanks for your feedback @neolit123 
Because public method `NewRestartConfig` returns an instance of private `restartDaemonConfig`.

Reply:
This is the linter error: @neolit123 
```
daemon_restart.go:71:110: exported func NewRestartConfig returns unexported type *apps.restartDaemonConfig, which can be annoying to use
```

Reply:
thanks for the info.
the change seems OK.